diff --git a/node_modules/human-object-diff/src/diff/utils/append-dot-path.js b/node_modules/human-object-diff/src/diff/utils/append-dot-path.js
index ee69110..ac6e7be 100644
--- a/node_modules/human-object-diff/src/diff/utils/append-dot-path.js
+++ b/node_modules/human-object-diff/src/diff/utils/append-dot-path.js
@@ -4,13 +4,16 @@ function appendDotPath(diff) {
     return diff.dotpath;
   }
 
-  let propertyIndex = diff.path.length - 1;
-  while (typeof diff.path[propertyIndex] !== 'string') propertyIndex -= 1;
-  diff.dotpath = diff.path
-    .slice(0, propertyIndex + 1)
-    .reduce(reducer(diff), '');
-
-  return diff.dotpath;
+  if (diff.path && diff.path.length > 0) {
+    let propertyIndex = diff.path.length - 1;
+    while (typeof diff.path[propertyIndex] !== 'string' && propertyIndex > 0) propertyIndex -= 1;
+    diff.dotpath = diff.path
+      .slice(0, propertyIndex + 1)
+      .reduce(reducer(diff), '');
+    return diff.dotpath;
+  } else {
+    return undefined;
+  }
 }
 
 function reducer(diff) {
